compiler/control/OMROptions.cpp

```diff
@@ -2328,6 +2332,23 @@ OMR::Options::jitLatePostProcess(TR::OptionSet *optionSet, void * jitConfig)
       self()->setOption(TR_ReservingLocks,false);
 #endif
 
+#if defined(TR_HOST_64BIT)
+#if defined(J9ZOS390)
+   // We allocate code cache memory on z/OS by asking the port library for typically small (~2MB) code cache chunks.
+   // This is done because the port library can typically only allocate executable memory (code caches) below the
+   // 2GB bar. When RMODE(64) is enabled however we are able to allocate executable memory above the 2GB bar. This
+   // means two code caches could be very far apart from one another which is something we want to avoid as it may
+   // cause performance issues. The solution is to enable the code cache consolidation which will allocate larger
+   // (~256MB) code caches at startup. The code caches will then have locality which improves performance, but
+   // moreover the size of the allocated larger code cache will ensure no trampolines are needed in JIT private
+   // linkage.
+   if (self()->getOption(TR_EnableRMODE64))
+#endif
+      {
+      self()->setOption(TR_EnableCodeCacheConsolidation);
```

Comment:
This will not work. TR_EnableCodeCacheConsolidation is used in onLoadInternal() when the code caches are created. At that point the command line options have been processed. However, jitLatePostProcess() is done at a later stage in aboutToBootstrap(). I suggest to move this code one phase earlier in jitPostProcess